hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/FilePerBlockStrategy.java

```diff
@@ -174,8 +175,25 @@ public void writeChunk(Container container, BlockID blockID, ChunkInfo info,
       ChunkUtils.validateChunkSize(channel, info, chunkFile.getName());
     }
 
-    ChunkUtils
-        .writeData(channel, chunkFile.getName(), data, offset, len, volume);
+    long fileLengthBeforeWrite;
+    try {
+      fileLengthBeforeWrite = channel.size();
+    } catch (IOException e) {
+      throw new StorageContainerException("Encountered an error while getting the file size for "
+          + chunkFile.getName(), CHUNK_FILE_INCONSISTENCY);
+    }
+
+    ChunkUtils.writeData(channel, chunkFile.getName(), data, offset, len, volume);
+
+    // When overwriting, update the bytes used if the new length is greater than the old length
+    // This is to ensure that the bytes used is updated correctly when overwriting a smaller chunk
+    // with a larger chunk at the end of the block.
+    if (overwrite) {
+      long fileLengthAfterWrite = offset + len;
```

Comment:
```suggestion
      long fileLengthAfterWrite = offset + chunkLength;
```
Renaming this variable above makes it clear that we are using the chunk length, not the file or block length in this comparison.